### ISSUE TITLE
Dates ignore sub-seconds in comparison

### ISSUE THREAD

**Issue by hiiamboris** [red/red#4855]
**Describe the bug**
```
>> t: now/precise
>> t': t + 0:0:0.01
>> t' <= t
== true
>> t' >= t
== true
```

**Expected behavior**

`t' <= t` to be false

**Platform version**
```
Red 0.6.4 for Windows built 2-Mar-2021/15:01:01+03:00 commit #6de9b93
```

**Comment by greggirwin** [comment]
```
>> t' = t
== true
>> t' <> t
== false
>> t' < t
== false
>> t' > t
== false
```

**Comment by greggirwin** [comment]
Date compare intentionally rounds (%date.reds @Line:1093). @dockimbel will have to say why.
```
		t1: floor value1/time + 0.5						;-- in UTC already, round to integer
		t2: floor value2/time + 0.5
```

**Comment by hiiamboris** [comment]
R2 didn't do that:
```
>> t: now/precise
>> t + 0:0:0.0000000001 = t
== true
>> t + 0:0:0.000000001 = t
== false
```

**Comment by qtxie** [comment]
Pushed a fix in PR #4858.
